### PR TITLE
feat: post metadata with the direct upload with skip verify

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-6696

### CODE REVIEW

**Comment by merlin-northern** [comment]
friendly deployments pull: https://github.com/mendersoftware/deployments/pull/931

**Comment by merlin-northern** [comment]
@tranchitella here as well, give me general greenlight, I have not even read it before asking for a review, so may need some cleaning up.

**Comment by merlin-northern** [comment]
> @tranchitella here as well, give me general greenlight, I have not even read it before asking for a review, so may need some cleaning up.

but you can use it to test:
```
./mender-cli artifacts --server https://somehost --skip-verify upload --direct /some/artifact
```

File: client/deployments/client.go [hunk 1]
```diff
@@ -199,6 +201,76 @@ func listArtifact(a artifactData, detailLevel int) {
...
+
+type DirectUploadMetadata struct {
+	Size    int64    `json:"size,omitempty" valid:"-"`
+	Updates []Update `json:"updates" valid:"-"`
+}
+
+func readArtifactMetadata(path string, size int64) io.Reader {
+	r, err := os.Open(path)
+	if err != nil {
+		return nil
```
Comment (tranchitella) [inline client/deployments/client.go:239]: Shouldn't we at least log the error message?

File: client/deployments/client.go [hunk 2]
```diff
@@ -199,6 +201,76 @@ func listArtifact(a artifactData, detailLevel int) {
...
+	}
+	defer r.Close()
+	ar := areader.NewReader(r)
+	err = ar.ReadArtifact()
+	if err != nil {
+		return nil
```
Comment (tranchitella) [inline client/deployments/client.go:245]: Shouldn't we at least log the error message?

File: client/deployments/client.go [hunk 3]
```diff
@@ -199,6 +201,76 @@ func listArtifact(a artifactData, detailLevel int) {
...
+			})
+		}
+	}
+	directMetadata := DirectUploadMetadata{
+		Size:    size,
+		Updates: directUploads,
+	}
+	data, err := json.Marshal(directMetadata)
+	if err != nil {
+		return nil
```
Comment (tranchitella) [inline client/deployments/client.go:269]: Shouldn't we at least log the error message?

**Review by tranchitella — APPROVED** [review]

**Pull request merged**